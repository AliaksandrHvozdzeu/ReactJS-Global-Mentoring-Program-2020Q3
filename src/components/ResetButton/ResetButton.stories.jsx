import React from 'react';
import { withKnobs } from '@storybook/addon-knobs';
import ResetButton from './js/ResetButton';
import '../../App.css';

export default {
  title: 'ResetButton',
  decorators: [withKnobs],
};

export const Default = () => (
  <ResetButton onReset={() => alert('Default')}/>
);
