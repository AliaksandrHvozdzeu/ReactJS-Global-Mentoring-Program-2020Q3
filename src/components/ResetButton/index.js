import ResetButton from './js/ResetButton';

export default ResetButton;
