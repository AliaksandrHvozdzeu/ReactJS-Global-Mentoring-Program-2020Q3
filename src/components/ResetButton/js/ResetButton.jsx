import React from 'react';
import PropTypes from 'prop-types';
import '../css/ResetButton.css';

const ResetButton = ({ onReset }) => {

  return (
    <>
      <button type="button"
              className="reset-button"
              onClick={onReset}>RESET
      </button>
    </>
  );

};

ResetButton.propTypes = {
  onReset: PropTypes.func.isRequired,
};

export default ResetButton;

