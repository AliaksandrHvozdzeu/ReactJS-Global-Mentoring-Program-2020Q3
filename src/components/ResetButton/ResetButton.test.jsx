import React from 'react';
import rerender from 'react-test-renderer';
import { jest } from "@jest/globals";
import { shallow } from "enzyme";
import ResetButton from "../ResetButton";

describe('when ResetButton', () => {

  const onReset = jest.fn();

  test('then snapshot created', () => {
    const component = rerender.create(<ResetButton onReset={onReset}/>);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('then reset button click', () => {
    const backResetButton = shallow(<ResetButton onReset={onReset}/>);
    backResetButton.simulate('click');
    expect(backResetButton.text()).toEqual("RESET");
  });

});
