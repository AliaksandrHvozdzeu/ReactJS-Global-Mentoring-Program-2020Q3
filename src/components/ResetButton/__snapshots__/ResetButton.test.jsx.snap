// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`when ResetButton then snapshot created 1`] = `
<button
  className="reset-button"
  onClick={[MockFunction]}
  type="button"
>
  RESET
</button>
`;
